bug: Error when running ergo without command

I am having an error when running ergo without providing a command.

Expected behavior:
Show USAGE

Actual behavior:
```
ergo
panic: runtime error: index out of range

goroutine 1 [running]:
main.command(0x12da13e)
        /Users/cristianoliveira/go/src/github.com/cristianoliveira/ergo/main.go:45 +0xc8c
main.main()
        /Users/cristianoliveira/go/src/github.com/cristianoliveira/ergo/main.go:127 +0xb5
```
The problem is here. We should look if `os.Args` has more than 1 item before creating the command
```go
func command() func() {
	config := proxy.NewConfig()
>>>	command := flag.NewFlagSet(os.Args[1], flag.ExitOnError)
	configFile := command.String("config", "./.ergo", "Set the services file")
	command.Parse(os.Args[2:])
```

It should be easy to solve.

**Please also provide a test for it**

--- a/main.go
+++ b/main.go
@@ -41,6 +41,11 @@ setup:
 `
 
 func command() func() {
+	// Fail fast if we didn't receive a command argument
+	if len(os.Args) == 1 {
+		return nil
+	}
+
 	config := proxy.NewConfig()
 	command := flag.NewFlagSet(os.Args[1], flag.ExitOnError)
 	configFile := command.String("config", "./.ergo", "Set the services file")
@@ -124,11 +129,16 @@ func main() {
 		return
 	}
 
-	cmd := command()
-	showUsage := *help || len(os.Args) == 1 || cmd == nil
-
-	if showUsage {
+	if *help {
 		fmt.Println(USAGE)
+		return
+	}
+
+	cmd := command()
+
+	if cmd == nil {
+		fmt.Println(USAGE)
+		os.Exit(1)
 	} else {
 		cmd()
 	}

--- a/main_test.go
+++ b/main_test.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"os"
+	"os/exec"
 	"testing"
 )
 
@@ -12,6 +13,28 @@ func TestMain(t *testing.T) {
 
 		main()
 		// Output: USAGE
+	})
+}
+
+// TestMissingCommand will cause the main function to return a non-zero exit code
+// so we need to do some wrapping to make the test not fail.
+// For more info check ou to uset: https://talks.golang.org/2014/testing.slide#23
+func TestMissingCommand(t *testing.T) {
+	t.Run("missing a command so it shows usage", func(tt *testing.T) {
+		if os.Getenv("TEST_MISSING_COMMAND") == "1" {
+			args := []string{"ergo"}
+			os.Args = args
+
+			main()
+			// Output: USAGE and exit with an error code
+		}
+		cmd := exec.Command(os.Args[0], "-test.run=TestMissingCommand")
+		cmd.Env = append(os.Environ(), "TEST_MISSING_COMMAND=1")
+		err := cmd.Run()
+		if e, ok := err.(*exec.ExitError); ok && !e.Success() {
+			return
+		}
+		t.Fatalf("process ran with err %v, want exit status 1", err)
 	})
 }
 

--- a/main_test.go
+++ b/main_test.go
@@ -6,6 +6,8 @@ import (
 	"testing"
 )
 
+var testExec = os.Args[0]
+
 func TestMain(t *testing.T) {
 	t.Run("it shows usage", func(tt *testing.T) {
 		args := []string{"ergo", "-h"}
@@ -28,7 +30,7 @@ func TestMissingCommand(t *testing.T) {
 			main()
 			// Output: USAGE and exit with an error code
 		}
-		cmd := exec.Command(os.Args[0], "-test.run=TestMissingCommand")
+		cmd := exec.Command(testExec, "-test.run=TestMissingCommand")
 		cmd.Env = append(os.Environ(), "TEST_MISSING_COMMAND=1")
 		err := cmd.Run()
 		if e, ok := err.(*exec.ExitError); ok && !e.Success() {

